packages/lodestar/src/api/impl/validator/validator.ts

```diff
@@ -221,6 +226,14 @@ export class ValidatorApi implements IValidatorApi {
       slot,
       committeeIndex
     );
+    const subnet = getCommitteeIndexSubnet(committeeIndex);
+    const peerIds = this.reps.getPeerIdsBySubnet(subnet);
+    if (peerIds.length < 3) {
+      // If an insufficient number of current peers are subscribed to the topic,
+      // the validator must discover new peers on this topic
+      this.logger.info(`Found only ${peerIds.length} for subnett ${subnet}, finding new peers to connect`);
+      await this.network.connectToNewPeersBySubnet(parseInt(subnet), peerIds);
+    }
```

Comment:
I feel like this whole logic should be moved to network. Like we should just call 
```
this.network.searchSubnetPeers(subnet)
```
here